Fix #19968: Note type search is case sensitive and can't be deleted quickly

Ayush-Patel-56 opened the pull request:
## Purpose / Description
Fixes two issues in the "Manage note types" screen:
1.  **Case Sensitivity:** The search filter was case-sensitive, making it hard to find types like "Basic" if typing "basic".
2.  **Typing Interruption:** The search input was being reset loosely on every state update, interrupting the soft keyboard loop and preventing the "long press delete" key from working effectively.

## Fixes
* Fixes #19968

## Approach
- Updated the search filter to use `ignoreCase = true`.
- Added a check to the `SearchView` binding to only update the query text if it has actually changed. This prevents the keyboard/input loop from being reset unnecessarily.

## How Has This Been Tested?
I tested this manually on an emulator:
1.  **Search Test:** Typed "bAsIc" (mixed case) and verified it correctly found "Basic" (see attached video).
2.  **Delete Test:** Typed a long string and long-pressed the backspace key. Verified that deletion is now rapid and smooth without interruption.

https://github.com/user-attachments/assets/e416a071-9f13-4372-81d6-f47252d476ec



## Checklist

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

david-allison commented:
Thanks! Please add a test

sanjaysargam commented on AnkiDroid/src/test/java/com/ichi2/anki/notetype/ManageNoteTypesViewModelTest.kt:
```diff
@@ -209,6 +209,22 @@ class ManageNoteTypesViewModelTest : RobolectricTest() {
+            val viewModel = ManageNoteTypesViewModel()
+            viewModel.filter(name.lowercase())
+            var currentlyDisplayed = viewModel.noteTypes.filter { it.shouldBeDisplayed }
+            assertEquals(1, currentlyDisplayed.size)
```
is it important? 
Can you explain this ?

Ayush-Patel-56 commented on AnkiDroid/src/main/java/com/ichi2/anki/notetype/ManageNotetypes.kt:
```diff
@@ -177,7 +177,9 @@ class ManageNotetypes : AnkiActivity(R.layout.activity_manage_note_types) {
                 findViewById<Toolbar>(R.id.toolbar).menu?.findItem(R.id.search_item)
             val searchView = searchMenuItem?.actionView as? AccessibleSearchView
-            searchView?.setQuery(state.searchQuery, false)
+            if (searchView?.query.toString() != state.searchQuery) {
```
added the comment here

sanjaysargam commented:
squash your commits into one

sanjaysargam commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-9.2.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.12.1-bin.zip
```
unwanted change. please undo this

Haz3-jolt commented:
@Ayush-Patel-56 please rewrite the PR to use our template, it is quite hard to otherwise, paste outputs of your test results but ideally screenshots or even a screen recording. 

It is hard to review without these.

Ayush-Patel-56 commented on AnkiDroid/src/test/java/com/ichi2/anki/notetype/ManageNoteTypesViewModelTest.kt:
```diff
@@ -159,6 +159,20 @@ class ManageNoteTypesViewModelTest : RobolectricTest() {
+                viewModel.state.value.noteTypes
+                    .filter { it.shouldBeDisplayed }
+            assertThat(currentlyDisplayed, hasSize(1))
+            assertEquals(testName, currentlyDisplayed[0].name)
```
Thanks! I see your point. Since you approved the PR, do you want me to push a commit to change that assertion style, or is it fine as-is for this PR?

david-allison commented:
`fix(ManageNoteTypes): make searching case-insensitive`

Having "deletion" there is unusual

david-allison commented:
commit message is permanent and should be designed for the long-term

We can and will fixup GitHub titles after the fact, but it should be edited as well. The title is better than the commit message title, so lower priority

Ayush-Patel-56 commented:
> commit message is permanent and should be designed for the long-term
> 
> We can and will fixup GitHub titles after the fact, but it should be edited as well. The title is better than the commit message title, so lower priority

Done! I've updated the commit message to be cleaner: fix(ManageNoteTypes): Make search case-insensitive and moved the deletion fix explanation to the body.
